Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -86,10 +86,14 @@ private class ParameterSetNames
 
         /// <summary>
         /// <para type="description">
-        /// The ID of the dataset that you want to get a descriptor object for.
+        /// The ID of the dataset that you want to get a descriptor object for. This field also accepts 
+        /// DatasetData objects so they can be mapped to full Dataset objects.
         /// </para>
         /// </summary>
-        [Parameter(Mandatory = true, Position = 0, ParameterSetName = ParameterSetNames.Get)]
+        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.Get)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(DatasetList.DatasetsData),
+            Property = nameof(DatasetList.DatasetsData.DatasetReference))]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(DatasetReference), Property = nameof(DatasetReference.DatasetId))]
         public string Dataset { get; set; }
```

Comment:
In that case though, the cmdlet will always use the default GCP project.  The user would have to specify that they are using another project with the -Project flag when they got the dataset object in the first place, so they should know to use the -Project flag again, or just to switch their default dataset while they're working in another location.